lxd/device/disk.go

```diff
@@ -157,6 +157,35 @@ func (d *disk) sourceIsLocalPath(source string) bool {
 	return true
 }
 
+// Check for inadequate sharing of block volumes.
+func (d *disk) checkBlockVolSharing(instanceType instancetype.Type, projectName string, volume *api.StorageVolume) error {
+	if instanceType == instancetype.Any {
+		return fmt.Errorf("Cannot add custom storage block volume to profile if security.shared is false or unset")
```

Comment:
But this function isn't checking security.shared so how can it know that? There's a layering violation here I think.

If this function is checking volume sharing as the name suggests, then it should check the volume's config right?